src/Collection.php

```diff
@@ -263,9 +263,30 @@ public function diffKeys(...$values): CollectionInterface
         return new self(DiffKeys::of()(...$values), $this->getIterator());
     }
 
-    public function distinct(): CollectionInterface
+    public function distinct(?callable $comparatorCallback = null, ?callable $accessorCallback = null): CollectionInterface
     {
-        return new self(Distinct::of(), $this->getIterator());
+        $accessorCallback ??=
+            /**
+             * @param T $value
+             * @param TKey $key
+             *
+             * @return T
+             */
+            static fn ($value, $key) => $value;
+
+        $comparatorCallback ??=
+            /**
+             * @param mixed $left
+             *
+             * @return Closure(mixed): bool
+             */
+            static fn ($left): Closure =>
+                /**
+                 * @param mixed $right
+                 */
+                static fn ($right): bool => $left === $right;
```

Comment:
would it not be cleaner to move these inside the `Distinct` operation?

Reply:
We could but I don't think I ever used nullable parameter in the operations (I might be wrong), this is why I preprocess them before calling the Distinct operation.

Reply:
but you already made it nullable here https://github.com/loophp/collection/pull/111/files?diff=split&w=1#diff-a8a263aeb0e5677f2d5480899096e5077a07ba66c37c6d45378d39f745b3f219R26

Reply:
Sure, in the collection, but not the operation itself.

Reply:
oops, you're right, I was looking at the wrong thing. Ok makes sense then let's keep it here then